salvo/src/lib/job_control_loader.py

```diff
@@ -13,9 +13,17 @@
 
 
 def _load_json_doc(filename):
+  """Load a disk file as JSON.
+
+  This function reads the specified filename and parses the contents
+  as JSON.
+
+  Args:
+      filename: The file whose contents are to be read as JSON data
+
+  Returns:
+      A dictionary representation of the file contents
```

Comment:
When describing a complex container, we should be more explicit about its structure and the inner object types to help the reader / user.

Example for this dictionary: {'John Doe': 345234}

Returns:
  A dict of strings to integers, the keys are names of people and the
  values are their phone numbers. Example:
    {'John Doe': 345234}

Including an example section is optional and I would only include it for more complex dictionaries.

Note - this is another requirement that goes away if we choose to use the pytype annotations.

This comment applies to a few locations.

Reply:
Can we revisit this point in a later PR. I have pytypes working locally and can add annotations to this module:

```
$ grep ^"def _load_json_doc" src/lib/job_control_loader.py
def _load_json_doc(filename: str) -> JobControl:
$ pytype src -P bazel-bin:.
Computing dependencies
Analyzing 25 sources with 6 local dependencies
ninja: Entering directory `/home/ubuntu/code/salvo/.pytype'
[31/31] check src.lib.service.test_s3_upload_event
Leaving directory '/home/ubuntu/code/salvo/.pytype'
Success: no errors found

```

Reply:
ptypes are coming.  I was able to get this going locally.

Reply:
Great news! Note that even with pytype, we should still say the meaning of the keys and values. Pytype removes the need to specifically mention the object types.

Something like:

```
Returns:
  A dictionary, keys are the file names and values are the file content.
```